### PR TITLE
[#38] Adding Black Formatting

### LINKED ISSUE #38: Format with Black
### What's Required?

As outlined in https://github.com/mackorone/spotify-playlist-archive/blob/master/TODO.md, the codebase should be formatted using Black.

**Acceptance Criteria**
* Code is formatted using black
* Black is added as a dependency to the project.

### PR DESCRIPTION
**Pull request by apoclyps**
### What's Changed

Apply's black formatting to the codebase

### Technical Description

* Removes unused imports
* Fixes typo in `Exception`
* Adds whitespace to break up sections
* Applies Black Formatting
* Creates requirements for the project

Closes #38

### CODE REVIEW

File: script.py [hunk 1]
```diff
@@ -538,7 +584,7 @@ def push_updates(now):
     print("Removing origin")
     remote_rm = run(["git", "remote", "rm", "origin"])
     if remote_rm.returncode != 0:
-        raise Excetion("Failed to remove origin")
+        raise Exception("Failed to remove origin")
```
Comment (mackorone) [inline script.py:556]: Good catch!

File: script.py [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
 import collections
 import datetime
-import json
 import os
 import re
-import requests
 import subprocess
 
+import requests
+
```
Comment (mackorone) [inline script.py:12]: I installed `requests` so long ago, I forgot it wasn't a builtin! Thanks for rearranging these.

**Review by mackorone — APPROVED** [review]
Looks good! It goes without saying, but *thank you* for breaking up your commits. Much easier to review!

**Review by mackorone — APPROVED** [review]
Looks good to me!

File: requirements_dev.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+click==7.1.2              # via black
+idna==2.10                # via requests
+mypy-extensions==0.4.3    # via black
+pathspec==0.8.0           # via black
+regex==2020.9.27          # via black
+requests==2.24.0          # via -r requirements.in
+toml==0.10.1              # via black
+typed-ast==1.4.1          # via black
+typing-extensions==3.7.4.3  # via black
+urllib3==1.25.10          # via requests
```
Comment (mackorone) [inline requirements_dev.txt:20]: Silly question: these comments were autogenerated, right?

File: requirements_dev.in [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Shared/Runtime dependencies
+
+-r requirements.in
+
+# Development-only dependencies
+
+
+# Lint-only dependencies
+
+black==20.8b1
```
Comment (mackorone) [inline requirements_dev.in:10]: Is this the standard `requirements_dev.in` template?

**Review by mackorone — APPROVED** [review]
This is going to sound silly, but I'm new to `requirements.in`. Can you give me a quick crash course or a pointer to some documentation? How is it different from `requirements.txt` and when should I use one versus the other?

Also, I searched for `requirements_dev.txt` and `requirements-dev.txt` was returned: https://pypi.org/project/requirements-dev.txt/. Is the latter more conventional?

**Comment by mackorone** [comment]
I did a bit more digging (https://github.com/jazzband/pip-tools). Is the difference between `requirements.in` and `requirements.txt` that the former is the list of direct dependencies, and the latter a list of transitive dependencies?

File: requirements_dev.txt (hunk 3 above)
Comment (apoclyps) [reply requirements_dev.txt]: All comment's within the requirements.txt files are autogenerated when running the corresponding `pip-compile requirements.in` command. As more transitive dependencies are added and more direct dependencies depending on a specific version, the direct dependency will also be added to the comment.

File: requirements_dev.in (hunk 4 above)
Comment (apoclyps) [reply requirements_dev.in]: Not quiet;

```
pip-compile --output-file=requirements-dev.txt requirements-dev.in
```

would produce 

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file=requirements-dev.txt requirements-dev.in
#
appdirs==1.4.4            # via black
black==20.8b1             # via -r requirements-dev.in
certifi==2020.6.20        # via requests
chardet==3.0.4            # via requests
click==7.1.2              # via black
idna==2.10                # via requests
mypy-extensions==0.4.3    # via black
pathspec==0.8.0           # via black
regex==2020.9.27          # via black
requests==2.24.0          # via -r requirements.in
toml==0.10.1              # via black
typed-ast==1.4.1          # via black
typing-extensions==3.7.4.3  # via black
urllib3==1.25.10          # via requests
``

However, adding `-r requirements.in` to the start of the dev dependencies avoids listing the entire contents of `requirements.txt`. 

The comments that appear after this line are not necessary (personal preference) but it helps provide a logical grouping to identify what dependencies belong too. Useful if/when more are added.

```
# Development-only dependencies


# Lint-only dependencies

black==20.8b1
```

I'm happy to make changes around any of this if it feels inappropriate for the project.

**Comment by apoclyps** [comment]
yes, that's correct. `requirements.in` for direct dependencies and `requirements.txt` for transitive dependencies. I'm happy to change this approach to a vanilla `requirements.txt`. 

I appreciate the project doesn't have very many dependencies and this approach may be more than what is necessary. I would like to know your thoughts :)

**Comment by mackorone** [comment]
@apoclyps thanks for the thoughtful responses. I think it's fine to keep this as is. There's no time like the present to adopt best practices! I'll go ahead and merge this.

**Pull request merged**